src/components/OutSegTemplateEditor.jsx

```diff
@@ -83,6 +84,24 @@ const OutSegTemplateEditor = memo(({ outSegTemplate, setOutSegTemplate, generate
 
   const needToShow = shown || error != null;
 
+  const onTextClick = useCallback(() => {
```

Comment:
not sure what's the purpose of this, so i'll remove it

Reply:
You're right, there is some code here in my original PR that didn't 100% work that I didn't notice before pushing. 

Overall, the code seems to work like how I expect after I pulled your changes; the user can still inject the button based on cursor position, even if the input focus is lost afterwards.

I checked your changes on my end here: https://i.imgur.com/bnX8Ds4.mp4

So, from my perspective everything is still working as I expect, and thanks for removing the extraneous code that doesn't do anything (to be honest, I used ChatGPT a little to help me out as I'm not _that_ familiar with ReactJS, so there may be some extra boilerplate sorry).